packages/core/admin/admin/src/content-manager/history/pages/History.tsx

```diff
@@ -91,42 +90,42 @@ const HistoryPage = () => {
     { refetchOnMountOrArgChange: true }
   );
 
-  // Make sure the user lands on a selected history version
-  React.useEffect(() => {
-    const versions = versionsResponse.data?.data;
-
-    if (!query.id && !versionsResponse.isLoading && versions?.[0]) {
-      navigate({ search: stringify({ ...query, id: versions[0].id }) }, { replace: true });
-    }
-  }, [versionsResponse.isLoading, navigate, query.id, versionsResponse.data?.data, query]);
-
-  const { permissions, isLoading: isLoadingPermissions } = useSyncRbac(slug!, query, 'History');
-
-  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading || isLoadingPermissions) {
-    return <Page.Loading />;
-  }
-
   /**
    * Ensure that we have the necessary data to render the page:
    * - slug for single types
    * - slug _and_ documentId for collection types
    */
-  if (!slug || (!documentId && collectionType === COLLECTION_TYPES)) {
+  if (!slug || (collectionType === COLLECTION_TYPES && !documentId)) {
     return <Navigate to="/content-manager" />;
   }
 
-  // TODO: real error state when designs are ready
-  if (versionsResponse.isError || !versionsResponse.data || !layout || !schema || !permissions) {
-    return null;
+  if (isLoadingDocument || isLoadingLayout || versionsResponse.isLoading) {
+    return <Page.Loading />;
+  }
+
+  // It was a success, handle empty data
+  if (
+    !versionsResponse.isError &&
+    (!versionsResponse.data || !versionsResponse.data?.data.length)
+  ) {
+    return <Page.NoData />;
   }
 
-  const selectedVersion = versionsResponse.data.data.find(
+  // We have data, handle selected version
+  if (versionsResponse.data?.data.length && !selectedVersionId) {
+    return (
+      <Navigate
+        to={`?${stringify({ ...query, id: versionsResponse?.data?.data[0].id })}`}
+        replace
+      />
+    );
+  }
```

Comment:
I'm not too keen on this whole navigation thing here in this page. I feel like we should know exactly where we are going when we click the button in the CM edit page.

Reply:
Ideally, yes, but since it would mean making another round trip on the server, I personally prefer what you have now

Reply:
I think you should also start using the object notation of `to` so you can just pass the `search` property and stop appending `?` 😄

Reply:
🤣 I knew that had to exist somewhere but I didn't look hard enough. My bad I'll update.